### PR TITLE
Build refactoring

### PR DESCRIPTION
**Pull request by jechague**
- Add GitHub Actions build for PRs and branch status
- Add dispatch for internal builds
- Protect workflows in CODEOWNERS

Please, do not mark as _Ready for Review_ until runners are installed

### CODE REVIEW

File: .github/updatePOMVersion.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+GIT_HASH=$(git rev-parse HEAD)
+if ! [ "$(git symbolic-ref --short HEAD)" = "beta" ]; then
+  echo "## Is MASTER or UPGRADE  branch, use pom.xml version"
+else
+  currentversion=$(sed -n '/<version>.*-SNAPSHOT<\/version>/ s/<version>\(.*\)-SNAPSHOT<\/version>/\1/p' pom.xml)
+  devversionname="$currentversion-${GIT_HASH:0:7}-SNAPSHOT"
+  echo "## Is DEVELOPER branch, update pom.xml version to: $devversionname"
+  mvn -B versions:set -DgenerateBackupPoms=false -DnewVersion="$devversionname"
+fi
```
Comment (jechague) [inline .github/updatePOMVersion.sh:11]: @anaiberta: this code is taken _as is_ from our internal build configuration but I'm not sure we should continue to use it. We no longer use the packages generated from PR builds so I think we could safely remove it. However, we could use this script to start using the master version + 100 in the beta branch. 

Could you also help me see whether I'm missing any special project configuration (eg. Mizuho, Corona, etc.) in this script?

File: .github/workflows/Build.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+on: 
+  pull_request:
+    branches:
+      - 'master'
+      - 'release-*'
+  push:
+    branches: 
+      - 'master'
+      - 'beta'
```
Comment (anaiberta) [inline .github/workflows/Build.yml:11]: 'beta-corona' is missed.

**Review by anaiberta — CHANGES_REQUESTED** [review]
Some things to review (after a first overview).

File: .github/workflows/Build.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      uses: actions/setup-java@v1.4.3
+      with:
+        java-version: 1.9
+
+    - name: Calculate build variables
+      id: buildVariables
+      run: |
+        if ! [[ "$GIT_REF" =~ 'release-[0-9]?\.[0-9]?$' ]]; then
+          CommitNumber=$(git rev-list --count HEAD)
+          NexusURL=${{ secrets.NEXUS_MAVEN_SNAPSHOTS }}
```
[the remark this answers is not in the document]
Comment (jechague) [reply .github/workflows/Build.yml]: This was leftover code from our internal build script. Fixed in 47e1852

File: .github/workflows/Build.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        else
+          CommitNumber=$(git rev-list --count origin/master..)
+          NexusURL=${{ secrets.NEXUS_MAVEN_RELEASES }}
+        fi
+
+        LastCommitter=$(git log -1 --pretty=format:%an)
+        CommitMessage=$(git log -1 --pretty=%B)
+
+        echo "::set-output name=IsPrerelease::$IsPrerelease"
```
[the remark this answers is not in the document]
Comment (jechague) [reply .github/workflows/Build.yml:49]: Fixed in 47e1852

File: .github/workflows/Build.yml (hunk 2 above)
Comment (jechague) [reply .github/workflows/Build.yml:11]: Fixed in 9468f48

**Comment by jechague** [comment]
The necessary changes to our internal build infrastructure are in place so we can start applying these changes to the beta branch

**Review by anaiberta — APPROVED** [review]

File: .github/updatePOMVersion.sh [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        newVersion="$pomMajorNumber.$pomMinorNumber.$pomPatchNumber"
+        ;;
+
+    *)
+        echo "## Is MASTER or feature branch, use pom.xml version"
+        ;;
+esac
+
+if [[ -n "${newVersion}" ]]; then
+    echo "## Updaing pom.xml version to: $newVersion"
```
Comment (anaiberta) [inline .github/updatePOMVersion.sh:48]:
```suggestion
    echo "## Updating pom.xml version to: $newVersion"
```

File: .github/updatePOMVersion.sh (hunk 1 above)
Comment (jechague) [reply .github/updatePOMVersion.sh]: This was fixed in c98a704

**Review by anaiberta — APPROVED** [review]

**Pull request merged**